subsys/bluetooth/mesh/dfu_srv.c

```diff
@@ -78,6 +78,8 @@ static void apply_rsp_sent(int err, void *cb_params)
 	struct bt_mesh_dfu_srv *srv = cb_params;
 
 	if (err) {
+		/* return phase back to give client one more chance. */
+		srv->update.phase = BT_MESH_DFU_PHASE_VERIFY_OK;
```

Comment:
Sorry, I missed that our target sample reboots right in the apply callback, so we need to store the state after the `if` but before the callback is called.